scripts/lean-depot.py

```diff
@@ -0,0 +1,293 @@
+                for v in os.listdir('.')
+                if (os.path.splitext(v)[1] == '.toml'
+                    and is_prefix_of('snapshot',v))}
+        return vs
+
+    finally:
+        os.chdir(dir)
+
+def load_snapshot(fn):
+    def to_list(v):
+        if isinstance(v, list): return v
+        else: return [v]
+    snap = toml.load(fn)
+    snap = { git : { 'name': k, 'rev': v['rev'] }
+             for (k,v) in snap['snapshot'].items()
+             for git in to_list(v['git']) }
+    return snap
+
+def find_best_match(prj):
+    tag   = prj['package']['lib_snapshot']
+    vers  = prj['package']['lean_version']
+    deps  = prj['dependencies']
+    depot = get_depot(vers)
+    depot = [ (k, load_snapshot(v))
+                  for (k,v) in depot.items() ]
+                          # key=lambda x: x[0])
+    depot = [ k
+              for (k,v) in depot
+              if len([ k2 for k2 in deps.values()
+                       if k2 not in v ]) == 0 ]
+    (x,depot) = (depot[0], depot[1:])
+    if len(depot) == 0: snap = x
+    else: snap = max(depot)
+    return snap
+
+def wrap_indent(m, t, n):
+    m = ''.ljust(m)
+    return ("\n".join([ m + x for x in textwrap.wrap(t, n)]))
+
+@user_command('configure')
+def configure():
+    """setup leanpkg configuration"""
+
+    prj = toml.load('lean-depot.toml')
+    tag  = prj['package']['lib_snapshot']
+    vers = prj['package']['lean_version']
+    depot = toml.load(get_depot(vers)[tag])
+    # print(depot)
+    # print(prj)
+    name         = prj['package']['name']
+    lean_version = prj['package']['lean_version']
+    snapshot = { v['git'][0] : v['rev'] for (k,v) in depot['snapshot'].items() }
+    deps = { k : inline_table({ 'git' : v, 'rev' : snapshot[v] })
+             for (k,v) in prj['dependencies'].items() }
+    deps.update( prj['extra_deps'].items()  )
+    leanpkg = { 'package' : { 'name' : name,
+                              'lean_version' : lean_version,
+                              'path' : 'src',
+                              'version' : '0.1' },
+                'dependencies' : deps }
+    with open('leanpkg.toml', 'w') as h:
+        h.write(toml.encoder.dumps(leanpkg, toml.encoder.TomlPreserveInlineDictEncoder()))
+        h.write('# generated by lean-depot\n')
+        h.write('#   * edit lean-depot.toml and not this file\n')
+    os.system('leanpkg configure')
+
+@user_command('init')
+def init():
+    """create lean-depot configuration from leanpkg file"""
+    leanpkg = toml.load(f'leanpkg.toml')
+    name         = leanpkg['package']['name']
+    lean_version = leanpkg['package']['lean_version']
+    lean_version = str.split(lean_version,':')[-1]
+    depot = get_depot(lean_version)
+    if len(depot) <= 1:
+        snapshot = 'nightly'
+    else:
+        snapshot = max(filter(lambda x: x != 'nightly', depot.keys()))
+    all_deps = leanpkg['dependencies']
+    deps = { k : v['git'][0] for (k,v) in all_deps.items()
```

Comment:
I don't understand this `[0]`. Isn't `v['git']` going to be something like `'https://github.com/leanprover/mathlib'`, so that `v['git'][0]` is `h`?